### PR TITLE
Adding messaging for open and closed states

### LINKED ISSUE #7203: Secondary settings button missing descriptive text
**Describe the bug**
The secondary settings button has "More" as its `aria-label`, but this does not describe what it represents and its action.

**To Reproduce**
1. Edit a page
2. View the markup of the three dotted settings icon beside the cog icon.

example

**Expected behavior**
* The label of the button should describe what it is and what it does.
* That text should change when the action changes (when it has been expanded).

For example, the label could be changed from "More" to "Show additional settings" and "Hide additional settings" when content has been expanded.

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser: Chrome
 - Version: Latest

### PR DESCRIPTION
**Pull request by ryanwelcher**
## Description
This PR adds more descriptive aria-labels for the open and closed states for the Secondary Settings button and fixes #7203

## How has this been tested?
I followed the steps to reproduce in the corresponding issue.

## Types of changes
Bug fix (non-breaking change which fixes an issue) 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

**Review by adamsilverstein — APPROVED** [review]
Looks good to me!

File: packages/edit-post/src/components/header/more-menu/index.js
Comment (ocean90) [inline packages/edit-post/src/components/header/more-menu/index.js]: There's no need for adding a context here since both strings are only used once.

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: packages/edit-post/src/components/header/more-menu/index.js
Comment (ryanwelcher) [reply packages/edit-post/src/components/header/more-menu/index.js]:
```suggestion
const ariaClosed = _x( 'Show additional settings');
```

**Comment by ryanwelcher** [comment]
@ocean90 I've pushed the requested changes.

**Review by ocean90 — APPROVED** [review]

**Comment by youknowriad** [comment]
e2e tests are failing (we're probably still refering to the old label in some tests).

@jasmussen Thoughts on these changes?

**Comment by jasmussen** [comment]
This is a cool PR, and I have no interface design wise issue with it, and it solves an issue.

I want to note that there was relatively great controversy when this menu was named. "More Menu" is what it's referred to in UI blueprints and documentation. 

But does this name need to be surfaced in the UI? The Block Library isn't surfaced as the "Block Library" anywhere in the UI, its help text says "Add block". So in that vein, this seems okay to me. 

However the phrasing seems like it could use a little tuning. That menu is not only for "Additional settings". It's for quite a few different things:

- Editor modes
- Accessing the code editor
- Plugin shortcuts
- Tools like "Copy all content"
- A link to a Keyboard Shortcuts cheatsheet
- A link to an Options modal

Screenshot:

screenshot 2018-10-26 at 13 22 58

As such, simply labelling this as "additional settings" seems imprecise. I defer to those of you better versed in aria intricacies than me, but this is a _menu_, and it is in part designed with Google Docs menus as its inspiration. In that vein, here's the Google Docs tools menu markup when it's opened and closed:

docs tools menu

The label does not change on open/close — not that i mind it doing so here — but worth noting regardless, in case it can help inspire a better label.

I would prefer we do not end up with something like "Show additional settings, tools, editing modes, help and options".

**Comment by jasmussen** [comment]
Perhaps one of these could work?

- More tools & options
- Help, tools & options
- Preferences & more
- Help & more
- Help, settings & more

**Comment by ryanwelcher** [comment]
@jasmussen thanks for the feedback. Personally, I like the first option.

Just to confirm this will look like the following

Closed state: `Show more tools & options`
Open state: `Hide more tools & options`

If everyone concerned is OK with that, I'm happy to make the change ASAP.

**Comment by ryanwelcher** [comment]
> e2e tests are failing (we're probably still refering to the old label in some tests).
@youknowriad I updated the snapshot and my local tests are passing - is there another step that I missed to get the tests passing?

**Comment by ryanwelcher** [comment]
As a follow-up, I believe I need to update the selectors in the test/e2e/spec directory to match the copy change we decide on here

For example the selector here - https://github.com/WordPress/gutenberg/blob/master/test/e2e/specs/nux.test.js#L139

**Comment by ryanwelcher** [comment]
I've made the copy change indicated and updated the tests instances.

**Review by youknowriad — APPROVED** [review]
LGTM 👍 Let's merge if the tests pass.

**Comment by mikedang** [comment]
> Closed state: `Show more tools & options`
> Open state: `Hide more tools & options`

The copy makes sense and looks good to me. I would avoid using the ampersand here — use "and" instead: "Show more tools and options" and "Hide more tools and options".

**Pull request merged**
